### PR TITLE
[RFR] Test cloud subnet details

### PR DESCRIPTION
**Pull request by aljazkosir**
This test creates sandbox with Nuage cloud tenant and its entities. We then navigate to Cloud tenant's subnets through provider, where we compare UI stats with sandbox stats. 
We are testing both L3 and L2 subnets. The difference is that, L3 subnets are always connected to network router while L2 subnets arent.

Depends on: https://github.com/ManageIQ/integration_tests/pull/8115

Please review second commit as the first is different PR.

\cc @miha-plesko

### CODE REVIEW

File: cfme/networks/subnet.py [hunk 1]
```diff
@@ -84,6 +121,12 @@ def provider(self):
         from cfme.networks.provider import NetworkProvider
         return parent_of_type(self, NetworkProvider)
 
+    @property
+    def parent_tenant(self):
```
Comment (miha-plesko) [inline cfme/networks/subnet.py]: We could cache it with `@cached_property` because later we call it twice in a row and it looks like an expensive operation

File: cfme/tests/networks/nuage/test_inventory.py [hunk 2]
```diff
@@ -18,3 +18,39 @@ def test_tenant_details(setup_provider_modscope, provider, with_nuage_sandbox_mo
...
     })
+
+
+def test_tenant_subnet_details(setup_provider_modscope, provider, with_nuage_sandbox_modscope):
+    sandbox = with_nuage_sandbox_modscope
+    tenant_name = sandbox['enterprise'].name
+    subnet_name = sandbox['subnet'].name
+    tenant = provider.collections.cloud_tenants.instantiate(name=tenant_name)
+    subnet = tenant.collections.subnets.instantiate(
+        name=subnet_name)
```
Comment (miha-plesko) [inline cfme/tests/networks/nuage/test_inventory.py]: This can be in previous line I think

File: cfme/tests/networks/nuage/test_inventory.py [hunk 3]
```diff
@@ -18,3 +18,39 @@ def test_tenant_details(setup_provider_modscope, provider, with_nuage_sandbox_mo
         'num_security_groups': 2,
         'num_network_ports': 4
     })
+
+
+def test_tenant_subnet_details(setup_provider_modscope, provider, with_nuage_sandbox_modscope):
```
Comment (miha-plesko) [inline cfme/tests/networks/nuage/test_inventory.py]: Please rename to `test_subnet_details`

File: cfme/tests/networks/nuage/test_inventory.py [hunk 4]
```diff
@@ -18,3 +18,39 @@ def test_tenant_details(setup_provider_modscope, provider, with_nuage_sandbox_mo
...
+        'cidr': '192.168.0.0/24',
+        'gateway': '192.168.0.1',
+        'network_protocol': 'ipv4',
+        'network_router': sandbox['domain'].name,
+        'num_network_ports': 2,
+        'num_security_groups': 0
+    })
+
+
+def test_tenant_subnet_l2_details(setup_provider_modscope, provider, with_nuage_sandbox_modscope):
```
Comment (miha-plesko) [inline cfme/tests/networks/nuage/test_inventory.py]: `test_l2_subnet_details`

File: cfme/tests/networks/nuage/test_inventory.py [hunk 5]
```diff
@@ -18,3 +18,39 @@ def test_tenant_details(setup_provider_modscope, provider, with_nuage_sandbox_mo
...
+    sandbox = with_nuage_sandbox_modscope
+    tenant_name = sandbox['enterprise'].name
+    subnet_name = sandbox['l2_domain'].name
+    tenant = provider.collections.cloud_tenants.instantiate(name=tenant_name)
+    subnet = tenant.collections.subnets.instantiate(
+        name=subnet_name)
```
Comment (miha-plesko) [inline cfme/tests/networks/nuage/test_inventory.py]: Same here, this could be in single line

File: cfme/networks/subnet.py (hunk 1 above)
Comment (aljazkosir) [reply cfme/networks/subnet.py]: I pushed new changes, this part of the code was changed as it was not working as I was expecting.

File: cfme/tests/networks/nuage/test_inventory.py [hunk 6]
```diff
@@ -18,3 +18,39 @@ def test_tenant_details(setup_provider_modscope, provider, with_nuage_sandbox_mo
         'num_security_groups': 2,
         'num_network_ports': 4
     })
+
+
+def test_subnet_details(setup_provider_modscope, provider, with_nuage_sandbox_modscope):
```
Comment (miha-plesko) [inline cfme/tests/networks/nuage/test_inventory.py]: Please add a docstring to inform user what the difference between L3 and L2 subnet is so he won't be scrathing head why we go via Router here, but not in the next test:

```python
"""
Ensure L3 subnet details displays expected info.

L3 subnets are always connected to routers, hence we navigate to them as
  Tenant > Router > Subnet
"""
```

File: cfme/tests/networks/nuage/test_inventory.py [hunk 7]
```diff
@@ -18,3 +18,39 @@ def test_tenant_details(setup_provider_modscope, provider, with_nuage_sandbox_mo
...
+        'cidr': '192.168.0.0/24',
+        'gateway': '192.168.0.1',
+        'network_protocol': 'ipv4',
+        'network_router': router_name,
+        'num_network_ports': 2,
+        'num_security_groups': 0
+    })
+
+
+def test_l2_subnet_details(setup_provider_modscope, provider, with_nuage_sandbox_modscope):
```
Comment (miha-plesko) [inline cfme/tests/networks/nuage/test_inventory.py]:
```python
"""
Ensure L2 subnet details displays expected info.

L2 subnets act as standalone and are thus not connected to any router.
We navigate to them as
  Tenant > Subnet
"""
```

File: cfme/networks/provider/__init__.py [hunk 8]
```diff
@@ -25,8 +25,7 @@
     OneProviderNetworkRouterView,
     OneProviderSecurityGroupView,
     OneProviderSubnetView,
-    NetworkProviderEditView
-)
+    NetworkProviderEditView)
```
Comment (miha-plesko) [inline cfme/networks/provider/__init__.py]: Is this change really needed here?

Comment (aljazkosir) [reply cfme/networks/provider/__init__.py]: No, sorry, this was overwritten by a rebase merge.

File: cfme/networks/views.py [hunk 9]
```diff
@@ -646,6 +646,34 @@ def is_displayed(self):
...
+    sidebar = View.nested(SubnetSideBar)
+    including_entities = View.include(SubnetEntities, use_parent=True)
+
+    @property
+    def is_displayed(self):
+        obj = self.context['object']
+        from cfme.networks.cloud_tenant import CloudTenant
+        from cfme.networks.network_router import NetworkRouter
+        is_tenant = isinstance(obj, CloudTenant)
+        is_router = isinstance(obj, NetworkRouter)
```
Comment (miha-plesko) [inline cfme/networks/views.py]: Nitpicking, but we can inline this simple enough conversation (`if isinstance(obj, CloudTenant): ...`)

**Comment by mshriver** [comment]
This includes commit from #8115, which should be reviewed and merged first. Marking WIPTEST.

**Comment by aljazkosir** [comment]
I fixed the conflicts so this one should also be ready for review :)
\cc @mshriver

**Comment by mshriver** [comment]
@aljazkosir Please sign your commits, and register your GPG key in your GH account so we see verified commits in your PRs. Thanks!

File: cfme/cloud/tenant.py [hunk 10]
```diff
@@ -430,3 +439,29 @@ def step(self, *args, **kwargs):
...
+
+@navigator.register(Tenant, 'CloudSubnets')
+class CloudSubnets(CFMENavigateStep):
+    VIEW = ParentWithSubnetView
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self):
+        item = 'Cloud Subnets'
+        if not int(self.prerequisite_view.entities.relationships.get_text_of(item)):
+            raise DestinationNotFound("This Cloud Tenant doesn't have {item}".format(item=item))
```
Comment (mshriver) [inline cfme/cloud/tenant.py]: I'm thinking this message (in both nav destinations) should include the tenant object as well, for reference.

Something like:
`raise DestinationNotFound('Cloud Tenant {} has a 0 count for {} relationships'.format(self.obj, item)`

File: cfme/networks/views.py [hunk 11]
```diff
@@ -646,6 +646,30 @@ def is_displayed(self):
                 self.title.text == '{name} (Summary)'.format(name=self.context['object'].name))
 
 
+class ParentWithSubnetView(BaseLoggedInPage):
```
Comment (mshriver) [inline cfme/networks/views.py]: Consider `NetworkEntitySubnetView` to be a bit more descriptive here?

Please update the docblock to be more complete - this is a view for use with a network entity that includes `SubnetCollection` in its `_collections`

Comment (aljazkosir) [reply cfme/networks/views.py]: Thanks for the name suggestion, I knew it was kinda bad :), but I didn't have any better ideas. I included this changes in new push.
